extensions/ql-vscode/src/model-editor/yaml.ts

```diff
@@ -209,19 +224,29 @@ function createDataExtensionYamlsByGrouping(
     const newMethods = newModeledMethods[method.signature];
     if (newMethods) {
       const filename = createFilename(method);
+      const canonicalFilename = canonicalizeFilename(filename);
 
       // Override any existing modeled methods with the new ones.
-      methodsByFilename[filename][method.signature] = [...newMethods];
+      methodsByCanonicalFilename[canonicalFilename][method.signature] = [
+        ...newMethods,
+      ];
+
+      if (!(canonicalFilename in actualFilenameByCanonicalFilename)) {
```

Comment:
This check should be unnecessary and the filename should always be present in `actualFilenameByCanonicalFilename` at this point. On line 194 we already loop through `methods` and populate the filenames.

It's good to be safe, but still I propose we remove this bit since we shouldn't need to update `actualFilenameByCanonicalFilename` here. If we want to be safer we could add some checks that throw an error if data isn't populated?

Reply:
Thanks, good spot. I don't think we need an error check here, so I'll just remove it.